app/controllers/search_controller.rb

```diff
@@ -4,6 +4,6 @@ def search
     @search = Listing.where(is_active: true)
     @search = @search.where(address: session[:address])
```

Comment:
Instead of using a `.where`, which will only try to find Listings with the exact location, please try the following:
```
@search.near(session[:address], 5, order: 'distance')
```